treatment-service/appcontext/appcontext.go

```diff
@@ -92,31 +93,33 @@ func NewAppContext(cfg *config.Config) (*AppContext, error) {
 		return nil, err
 	}
 
-	log.Println("Initializing message queue subscriber...")
 	var messageQueueService messagequeue.MessageQueueService
-	switch cfg.MessageQueueConfig.Kind {
-	case common_mq_config.NoopMQ:
+	if (cfg.PollerConfig != nil && cfg.PollerConfig.Enabled) || cfg.MessageQueueConfig.Kind == common_mq_config.NoopMQ {
```

Comment:
As mentioned in other comment, if you use pointer for `MessageQueueConfig` you'll need to do nilcheck everywhere to prevent panic. In this case
```
if (cfg.PollerConfig != nil && cfg.PollerConfig.Enabled) || (cfg.MessageQueueConfig != nil && cfg.MessageQueueConfig.Kind == common_mq_config.NoopMQ)
```

Reply:
This is fixed now.